simpleflow/canvas.py

```diff
@@ -4,6 +4,20 @@
 from .task import ActivityTask
 
 
+def propagate_attribute(obj, attr, val):
+    if isinstance(obj, Activity):
+        setattr(obj, attr, val)
+    elif isinstance(obj, ActivityTask):
+        setattr(obj.activity, attr, val)
+    elif isinstance(obj, Group):
+        for activities in obj.activities:
+            propagate_attribute(activities, attr, val)
+    elif isinstance(obj, FuncGroup):
+        setattr(obj, attr, val)
+    else:
+        raise Exception('Object not found {}'.format(type(obj)))
```

Comment:
Something more descriptive like: `Cannot propagate attribute for unknown type: {}` ?